deckhandler.c

```diff
@@ -79,45 +79,27 @@ deck_init_dh (st_deck_dh *deck_dh)
 void
 deck_shuffle_dh (st_deck_dh *deck_dh)
 {
-  int j = 0;
-  bool status[CARDS_IN_DECK];
-
-  do
-  {
-    status[j++] = 1;
-  }while (j < CARDS_IN_DECK);
-
-  j = 0;
-
-  int card = 0;
-
-  st_deck_dh temp_deck;
-
-  while (j < CARDS_IN_DECK)
+  int x;
+  for (x = 0; x < 3; x++)
   {
-    /* card will a random number between 0 and 51 */
-    card = rand () % CARDS_IN_DECK;
-
-    if (status[card])
+    int i = 0;
+    while (i < CARDS_IN_DECK)
```

Comment:
Something to remember We're effectively doing a lot more shuffles now because we switch 2 cards. But also, a card may get shuffled more than once. (I think this is fine.)

Reply:
It's fine as far as it seems to work. But as for efficiency, there may still be room for improvement.

I think it's okay if a card gets shuffled more than once, because that happens when shuffling in real life.

To get really real, maybe if the deck was split into two parts first.. and a different type of algorithm was applied... Can't picture that yet though. I may give that some thought.